#! /bin/sh
patch -p0 -f $* < $0
exit $?

[sm]
2009-07-30  Werner Koch  <wk@g10code.com>

	* call-agent.c (learn_cb): Do not store as ephemeral.



--- sm/call-agent.c     (revision 5101)
+++ sm/call-agent.c     (working copy)
@@ -875,13 +875,11 @@
       return 0;
     }
 
+  /* We do not store a certifciate with missing issuers as ephemeral
+     because we can assume that the --learn-card command has been used
+     on purpose.  */
   rc = gpgsm_basic_cert_check (parm->ctrl, cert);
-  if (gpg_err_code (rc) == GPG_ERR_MISSING_CERT)
-    { /* For later use we store it in the ephemeral database. */
-      log_info ("issuer certificate missing - storing as ephemeral\n");
-      keydb_store_cert (cert, 1, NULL);
-    }
-  else if (rc)
+  if (rc && gpg_err_code (rc) != GPG_ERR_MISSING_CERT)
     log_error ("invalid certificate: %s\n", gpg_strerror (rc));
   else
     {


